### PR TITLE
[Tests] Bump language used in the mac binding tests.

### PR DESCRIPTION
**Pull request by mandel-macaque**
They broke with nullability, we need to use the latests lang version.

Fixes: https://github.com/xamarin/xamarin-macios/issues/8346

### CODE REVIEW

File: tests/mac-binding-project/MobileBinding/MobileBinding_static.csproj [hunk 1]
```diff
@@ -9,6 +9,7 @@
     <RootNamespace>MobileBinding</RootNamespace>
     <MacResourcePrefix>Resources</MacResourcePrefix>
     <AssemblyName>MobileBinding</AssemblyName>
+    <BaseIntermediateOutputPath>..\obj\Mobile-static</BaseIntermediateOutputPath>
```
Comment (whitneyschmidt) [inline tests/mac-binding-project/MobileBinding/MobileBinding_static.csproj:12]: Is this related to updating the `LangVersion`?

Comment (mandel-macaque) [reply tests/mac-binding-project/MobileBinding/MobileBinding_static.csproj:12]: hmm good point, I bet it was vsmac trying to be smart :/
